storage/range.go

```diff
@@ -1370,3 +1402,119 @@ func (r *Range) AdminSplit(args *proto.AdminSplitRequest, reply *proto.AdminSpli
 		reply.SetGoError(util.Errorf("split at key %q failed: %s", splitKey, err))
 	}
 }
+
+// InitialState implements the raft.Storage interface.
+func (r *Range) InitialState() (raftpb.HardState, raftpb.ConfState, error) {
+	data, err := r.rm.Engine().Get(
+		proto.EncodedKey(engine.RaftStateKey(uint64(r.Desc.RaftID))))
+	if err != nil {
+		return raftpb.HardState{}, raftpb.ConfState{}, err
+	}
+	var hs raftpb.HardState
+	err = gogoproto.Unmarshal(data, &hs)
+	if err != nil {
+		return raftpb.HardState{}, raftpb.ConfState{}, err
+	}
+
+	// TODO(bdarnell): load ConfState if we have a previous snapshot.
```

Comment:
Add a TODO about getting conf state from the range metadata. We probably want to add transformers to/from RangeMetadata from raftpb.ConfState.